go/test/endtoend/recovery/pitr/shardedpitr_test.go

```diff
@@ -0,0 +1,531 @@
+	require.NoError(t, err)
+
+	// Start keyspace
+	keyspace := &cluster.Keyspace{
+		Name: keyspaceName,
+	}
+	clusterInstance.Keyspaces = append(clusterInstance.Keyspaces, *keyspace)
+
+	shard := &cluster.Shard{
+		Name: shardName,
+	}
+	shard0 := &cluster.Shard{
+		Name: shard0Name,
+	}
+	shard1 := &cluster.Shard{
+		Name: shard1Name,
+	}
+
+	// Defining all the tablets
+	master = clusterInstance.NewVttabletInstance("replica", 0, "")
+	replica = clusterInstance.NewVttabletInstance("replica", 0, "")
+	shard0Master = clusterInstance.NewVttabletInstance("replica", 0, "")
+	shard0Replica = clusterInstance.NewVttabletInstance("replica", 0, "")
+	shard1Master = clusterInstance.NewVttabletInstance("replica", 0, "")
+	shard1Replica = clusterInstance.NewVttabletInstance("replica", 0, "")
+
+	shard.Vttablets = []*cluster.Vttablet{master, replica}
+	shard0.Vttablets = []*cluster.Vttablet{shard0Master, shard0Replica}
+	shard1.Vttablets = []*cluster.Vttablet{shard1Master, shard1Replica}
+
+	clusterInstance.VtTabletExtraArgs = append(clusterInstance.VtTabletExtraArgs, commonTabletArg...)
+	clusterInstance.VtTabletExtraArgs = append(clusterInstance.VtTabletExtraArgs, "-restore_from_backup", "-enable_semi_sync")
+
+	err = clusterInstance.LaunchCluster(keyspace, []cluster.Shard{*shard, *shard0, *shard1})
+	require.NoError(t, err)
+	// Start MySql
+	var mysqlCtlProcessList []*exec.Cmd
+	for _, shard := range clusterInstance.Keyspaces[0].Shards {
+		for _, tablet := range shard.Vttablets {
+			proc, err := tablet.MysqlctlProcess.StartProcess()
+			require.NoError(t, err)
+			mysqlCtlProcessList = append(mysqlCtlProcessList, proc)
+		}
+	}
+
+	// Wait for mysql processes to start
+	for _, proc := range mysqlCtlProcessList {
+		err = proc.Wait()
+		require.NoError(t, err)
+	}
+
+	queryCmds := []string{
+		fmt.Sprintf("CREATE USER '%s'@'%%';", mysqlUserName),
+		fmt.Sprintf("GRANT ALL ON *.* TO '%s'@'%%';", mysqlUserName),
+		fmt.Sprintf("GRANT GRANT OPTION ON *.* TO '%s'@'%%';", mysqlUserName),
+		fmt.Sprintf("create database %s;", "vt_ks"),
+		"FLUSH PRIVILEGES;",
+	}
+
+	for _, tablet := range []*cluster.Vttablet{master, replica, shard0Master, shard0Replica, shard1Master, shard1Replica} {
+		for _, query := range queryCmds {
+			if _, err = tablet.VttabletProcess.QueryTablet(query, keyspace.Name, false); err != nil {
+				log.Errorf("error creating database for keyspace %v: %v", keyspace.Name, err)
+				return err
+			}
+		}
+		log.Infof("Starting vttablet for tablet uid %d, grpc port %d", tablet.TabletUID, tablet.GrpcPort)
+
+		if err = tablet.VttabletProcess.Setup(); err != nil {
+			log.Errorf("error starting vttablet for tablet uid %d, grpc port %d: %v", tablet.TabletUID, tablet.GrpcPort, err)
+			return err
+		}
+	}
+
+	err = clusterInstance.VtctlclientProcess.InitShardMaster(keyspaceName, shard.Name, cell, master.TabletUID)
+	require.NoError(t, err)
+
+	// Start vtgate
+	err = clusterInstance.StartVtgate()
+	require.NoError(t, err)
+
+	return nil
```

Comment:
as we already checked the errors, we don't need to return nil and check in main fnc

Reply:
We have return statement on line no 423 and 430, so we need the return variables for the function initializeCluster